app/api/processingnodes.py

```diff
@@ -35,4 +39,43 @@ class ProcessingNodeViewSet(viewsets.ModelViewSet):
 
     pagination_class = None
     serializer_class = ProcessingNodeSerializer
-    queryset = ProcessingNode.objects.all()
\ No newline at end of file
+    queryset = ProcessingNode.objects.all()
+
+
+class ProcessingNodeOptionsView(APIView):
+    """
+    Display the common options available among all online processing nodes. This is calculated by intersecting the available_options field of all online processing nodes visible to the current user.
+    """
+
+    queryset = ProcessingNode.objects.all()
+
+    def get(self, request):
+
+        nodes = get_objects_for_user(request.user, 'view_processingnode', ProcessingNode, accept_global_perms=False)
+        common_options = []
+
+        for node in nodes:
+            # Skip offline nodes
+            if not node.is_online():
+                continue
+
+            # First? Just populate
+            if len(common_options) == 0 and len(node.available_options) > 0:
+                common_options = node.available_options
+            else:
+                # Remove all options that are in common_options,
+                # but that are not in node.available_options
+                for common_option in common_options:
+                    found = False
+                    for option in node.available_options:
+                        if common_option['name'] == option['name']:
+                            found = True
+                            break
+
+                    # Mark for deletion
+                    if not found:
+                        common_option['_delete'] = True
+
+        common_options = [co for co in common_options if not '_delete' in co]
```

Comment:
Size is under control, since its the number of command line options times the number of processing nodes.

So O(command line options ^ 2 * processing nodes).

But all numbers are pretty small.